backend/src/config/database.config.ts

```diff
@@ -46,8 +78,56 @@ const databaseSchema = Joi.object({
     then: Joi.number(),
     otherwise: Joi.optional(),
   }).label('HD_DATABASE_PORT'),
+  ssl: Joi.when('needsSSL', {
+    is: Joi.valid('1'),
+    then: Joi.required(),
+    otherwise: Joi.optional(),
+  }).when('type', {
+    is: Joi.valid(DatabaseType.MARIADB, DatabaseType.MYSQL),
+    then: sqlTlsSchema,
+    otherwise: Joi.when('type', {
+      is: Joi.invalid(DatabaseType.SQLITE),
+      then: postgresTlsSchema,
+      otherwise: Joi.optional(),
+    }),
+  }),
 });
 
+const getTlsConfig = (dbType: DatabaseType, needsSSL: boolean) => {
+  if (!needsSSL) return;
+
+  const postgresTlsConfig = dbType === DatabaseType.POSTGRES && {
+    ssl: {
+      ca: getSecret(process.env.HD_PG_SSL_CA_PATH),
+      rejectUnauthorized: process.env.HD_PG_SSL_REJECT_UNAUTHORIZED === 'true',
+      key: getSecret(process.env.HD_PG_SSL_KEY_PATH),
+      cert: getSecret(process.env.HD_PG_SSL_CERT_PATH),
+    },
+  };
+
+  const sqlTlsConfig = [DatabaseType.MARIADB, DatabaseType.MYSQL].includes(
+    dbType,
+  ) && {
+    ssl: {
+      ca: getSecret(process.env.HD_SQL_SSL_CA_PATH),
+      key: getSecret(process.env.HD_SQL_SSL_KEY_PATH),
+      cert: getSecret(process.env.HD_SQL_SSL_CERT_PATH),
+      rejectUnauthorized: process.env.HD_SQL_SSL_REJECT_UNAUTHORIZED === 'true',
+      ciphers: process.env.HD_SQL_SSL_CIPHERS,
+      maxVersion: process.env.HD_SQL_SSL_MAX_VERSION,
+      minVersion: process.env.HD_SQL_SSL_MIN_VERSION,
+      passphrase: process.env.HD_SQL_SSL_PASSPHRASE,
+    },
+  };
+
+  return sqlTlsConfig || postgresTlsConfig;
+};
```

Comment:
As this method is rather crucial to database security options, it should be unit-tested. The `getSecret` method could be mocked for that.